### ISSUE TITLE
fripp: aggregation and default prefix are the same for flux_t handle

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#3833]
Problem: fripp aggregation policy and default metric prefix are per fripp context, and there is one fripp context per flux handle.  However in the broker, job manager, and job shell, there are multiple disjoint subsystems sharing a flux_t handle.

Example: there are existing stats in the broker's content cache use the default aggregation, but the stats being added to broker's overlay code should require  no aggregation.  Calling `flux_stats_set_period()` in the overlay code would change the policy in the content cache code.

I wonder if it would be better to expose the fripp context and functions directly for those cases?   E.g. each  component in that situation could explicitly create a fripp context and use it directly rather than the one in the handle?

Another idea would be to let the aggregation policy be set per metric.  (or -1 to use the handle default?)

**Comment by grondo** [comment]
Good observation. I'm not sure if exposing the fripp context and functions in the API is the right approach.
Two other ideas that come to mind:

1. Change the Flux "stats" API to operate on a `flux_stats_t` object which has to be created, with a prefix, from a flux handle, e.g.
 ```C
flux_stats_t *flux_stats_create (flux_t *h, const char *prefix, double interval); 
```
 Then the other `flux_stats_*` API calls take a `flux_stats_t` instead of `flux_t`.
2. Add a `char *prefix` argument to all API calls and internally create a fripp context for each prefix with a key in the aux hash which contains the prefix.

There are benefits and drawbacks to each approach.

**Comment by garlick** [comment]
Your first suggestion seems pretty reasonable.  Possibly the low level bits (like the parsed server address) could still be in a per-handle shared context, to avoid duplication. 

To reduce the boiler plate required to do stats, there could also be a `flux_stats_get()` function that creates or returns the handle from a flux_t handle, so you could just call e.g  `flux_stats_gauge_set (flux_stats_get (h), ...)`

**Comment by grondo** [comment]
Yeah, it is unfortunately not as clean as I'd hoped. (I think it would need to be `flux_stats_get (h, prefix)`). Maybe we can do some magic with macros to make the common case (a single prefix in a single flux_t handle) very simple to use.

**Comment by garlick** [comment]
More thought required maybe.
